### PR TITLE
add random playback button

### PR DESCRIPTION
**Pull request by FreaxMATE**
This adds a random playback button.
Peek 2020-01-23 16-44
It only appears when listening to more than one track. However, this is does **not shuffle** the playlist it just plays a random track.
In the future we should either make a preferences option to select shuffle/random or just using one method or setting up two buttons. I'd prefer the first method maybe with shuffle preselected.

FreaxMATE

### CODE REVIEW

**Comment by mbkma** [comment]
Nice job :) I'd prefer random playback to not be enabled be default though.

**Comment by FreaxMATE** [comment]
> Nice job :) I'd prefer random playback to not be enabled be default though.

That #

File: data/org.github.FreaxMATE.Ojo.gschema.xml [hunk 1]
```diff
@@ -39,6 +39,11 @@
       <summary>If enabled, coverart will be drawn, when available</summary>
       <description>Toggle borders</description>
     </key>
+    <key name='random-playback' type='b'>
+      <default>false</default>
```
Comment (FreaxMATE) [inline data/org.github.FreaxMATE.Ojo.gschema.xml:43]: Random playback is disabled by default.

**Review by FreaxMATE — COMMENTED** [review]
> Nice job :) I'd prefer random playback to not be enabled be default though.

That's actually the case.

**Comment by mbkma** [comment]
Icons work now in Adwaita, except the one for "no repeat".
When I listen to a track with "random playback" enabled, then on the next track select "no random playback" the following track is skipped for me.

**Comment by FreaxMATE** [comment]
> Icons work now in Adwaita, except the one for "no repeat".
> When I listen to a track with "random playback" enabled, then on the next track select "no random playback" the following track is skipped for me.

So this is not related to the new random-playback-mode it happened to me before. I think `libvlc_media_player_get_state()` in `ojo_player_end_reached()` in file ojo-player.c is maybe sending true although playback has already gone on.

**Review by mbkma — APPROVED** [review]
Good job!

**Pull request merged**